lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -284,67 +284,71 @@ def filter_stream(tag, es)
     end
 
     def filter_stream_from_files(tag, es)
-      new_es = MultiEventStream.new
```

Comment:
how about a short circuit like this:
```
    if es.empty?
      return es
    end
```
Same with filter_stream_from_journal

Also, please add a test for this.

Thanks!

Reply:
@richm certainly. Apologies for the noob question. I am struggling to see how to emit an empty EventStream in the test class that would trigger the short circuit. Can you give me any pointers?
--



@richm certainly. Apologies for the noob question. I am struggling to see how to emit an empty EventStream in the test class that would trigger the short circuit. Can you give me any pointers?
--

Reply:
not sure - MultiEventStream.new creates an empty stream?

Reply:
Yes correct, I am just trying to figure out how to call emit() and have it return an empty stream:

```
def emit(record, time = Engine.now)
  emit_with_tag(@tag, record, time)
end

def emit_with_tag(tag, record, time = Engine.now)
  @events[tag] ||= MultiEventStream.new
  @events[tag].add(time, record)
end
```
`emit_with_tag` always creates a new stream and adds the record.  So I cannot see how to make it empty.

Reply:
hmm - I have no idea - which makes me wonder how it is possible to run into this situation in the first place . . .

Reply:
unless you can figure out how to test this, I guess we just won't have a test for this case - I'm not sure how to test with an empty event stream

so just put the short circuit return in the filter_stream_from_files method.  Looks like filter_stream_from_journal is ok

Reply:
I beat my head against the wall trying to find a way to test this but I can't.  At least this adds a safety check to ensure we do not do any invalid options.  Updated with the requested changes.  You were right, journal did not need a change at all.